### ISSUE TITLE
[Insiders]: Typescript Syntax highlight is broken

### ISSUE THREAD

**Issue by limbosounds** [microsoft/vscode#89212]
Issue Type: <b>Bug</b>

1. Open typescript file
2. See that syntax highlight is broken

However, if I'm switching custom token color for `scope: entity` then any white token gets color of this customization.

Without token color customization
![Screenshot from 2020-01-24 10-22-39](https://user-images.githubusercontent.com/26549754/73054795-fcc36580-3e93-11ea-96e1-1335085cf3b6.png)

With token color customization:
![Screenshot from 2020-01-24 10-27-18](https://user-images.githubusercontent.com/26549754/73054893-31cfb800-3e94-11ea-8089-56049dd79c28.png)
![Screenshot from 2020-01-24 10-27-02](https://user-images.githubusercontent.com/26549754/73054899-35633f00-3e94-11ea-9df0-5a3e45959245.png)

VS Code version: Code - Insiders 1.42.0-insider (c4d53a19df8799f5d49a4f49a6900c499e6fa27c, 2020-01-23T06:38:09.164Z)
OS version: Linux x64 5.3.0-26-generic snap

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 2600 Six-Core Processor (12 x 1477)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: disabled_software<br>flash_stage3d: disabled_software<br>flash_stage3d_baseline: disabled_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: disabled_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: disabled_software<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|1, 1, 1|
|Memory (System)|31.38GB (12.20GB free)|
|Process Argv|--force-user-env --no-sandbox --unity-launch --no-sandbox|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
todo-tree|Gru|0.0.166
sass-indented|syl|1.7.3

</details>

**Comment by limbosounds** [comment]
BTW, it became broken within last 10 hours, because yesterday everything was fine

**Comment by markovicdenis** [comment]
Broken syntax highlighting here as well, don't know if it's related to your issue, but here is the error log from chrome dev tools...

` ERR Missing color in color map - #000000: Error: Missing color in color map - #000000
    at ColorMap.getId (/Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/node_modules.asar/vscode-textmate/release/main.js:357:19)
    at resolveParsedThemeRules (/Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/node_modules.asar/vscode-textmate/release/main.js:323:81)
    at Function.Theme.createFromParsedTheme (/Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/node_modules.asar/vscode-textmate/release/main.js:381:16)
    at Function.Theme.createFromRawTheme (/Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/node_modules.asar/vscode-textmate/release/main.js:378:21)
    at Registry.setTheme (/Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/node_modules.asar/vscode-textmate/release/main.js:3663:51)
    at o.setTheme (file:///Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4040:323)
    at C._doUpdateTheme (file:///Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4925:634)
    at C._doUpdateTheme (file:///Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4932:888)
    at C._updateTheme (file:///Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4925:545)
    at C._getOrCreateGrammarFactory (file:///Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4924:277)
    at async file:///Users/Shared/Apps/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:4924:625`

**Comment by Tapokj** [comment]
The same for pure JavaScript. Also sometimes the highlight returns to normal, secondarily, as if "flickering"

**Comment by limbosounds** [comment]
One more thing: highlighting is normal initially, when opening VSCode Insiders, but becomes broken after like 1-2 seconds. And after changing the settings of token color customizations it flickers to normal for like 0.5 seconds and becomes broken again :)

**Comment by cpheps** [comment]
I've noticed that comment coloring is not applying either when its overridden from the theme normal.

```
"[Chocolate Ice Cream]": {
            "comments": "#d689b6f6",
        }
```

I can change the `comments` color value and nothing changes.

**Comment by Raydir** [comment]
Same for individual Highlightings like e.g. contributed in my extension "Opacc F-Fscript".

before latest update:
![image](https://user-images.githubusercontent.com/17499573/73070085-bb40b380-3eaf-11ea-8b5d-b0f920e35c7b.png)

after latest update:
![image](https://user-images.githubusercontent.com/17499573/73070118-ce538380-3eaf-11ea-93c5-fe46e70a0937.png)

**Comment by aeschli** [comment]
Just to explain:
Semantic highlighting comes in once the typescript server is up.
Identifiers are then highlighted based on their actual type: types, interfaces, enums, namespace, parameters instead of just variables.
It's still work in progress and can be turned off with
`"editor.semanticHighlighting.enabled": false`

@limbosounds The rule
`"scope": "entity" is too broad. Basically every symbol is an entity: types, interfaces, functions. Grammars might coose other scopes, but that's what is used for the semantic tokens.
The `Developer: Show Editor Tokens and Scopes` shows semantic tokens and the rules that are probed for.

@marchenko-am That's an interesting issue. What theme do you use? Any customisation in the settings? Would you mind fining a new issue if you have more information?

@cpheps Thanks, good catch, fixed by 13dd07b8a811d61b9b1f068a0093be915dff4d0e

**Comment by limbosounds** [comment]
@aeschli so, it won't be fixed, lol? It was ok in previous version.

BTW, `scope.entity` works fine for my theme (Monokai Dimmed) - it replaces all red and dark-red tokens with my color.

**Comment by aeschli** [comment]
@limbosounds If you can paste me your source sample then I can investigate more.
But when you configure  `"scope": "entity" : red` that means that you want all types, namespaces & functions to be red. Now with semantic highlighting we actually know that the identifiers and not variables but types, namespaces & functions... That's why things look different.

**Comment by limbosounds** [comment]
@aeschli as you can see in OP, it does not work even without token customization.
I can share you copy-pasted code from screenshot here, if it will help :)

**Comment by aeschli** [comment]
@limbosounds Sorry, I didn't realize that 'Without token color customization' is also broken. I don't know your theme. What is the expected result?
Please make sure to update to the latest i-build (1.42.0-insider, 35116fa958d6622df1c31b4d84810b999a858170). There has been some fixes.

**Comment by limbosounds** [comment]
In c4b6a6b looks better (token color customizations are switched off, semantic highlight enabled yet):

![Screenshot from 2020-01-28 22-31-10](https://user-images.githubusercontent.com/26549754/73302825-f3ccee00-421d-11ea-97e8-4e6206383513.png)

but still some tokens are missing the color, as you can see. The expected result (from stable VSCode):

![Screenshot from 2020-01-28 22-32-45](https://user-images.githubusercontent.com/26549754/73302921-1fe86f00-421e-11ea-99d0-0ef20166fb8e.png)

For now I'm waitin' to be able to update, because c4b6a6b is supposed to be the latest available update for now ("There are no currently updates available")

**Comment by aeschli** [comment]
Can you set the cursor on the tokens with the wrong color and use the `Inspect Editor Token and Scopes`? That helps me to see where the color comes from.

**Comment by limbosounds** [comment]
@aeschli 
![Screenshot from 2020-01-29 14-31-15](https://user-images.githubusercontent.com/26549754/73357461-19510a80-42a5-11ea-9276-74719c718db0.png)
![Screenshot from 2020-01-29 14-35-06](https://user-images.githubusercontent.com/26549754/73357462-19e9a100-42a5-11ea-9c29-85971dd01217.png)
![Screenshot from 2020-01-29 14-35-58](https://user-images.githubusercontent.com/26549754/73357465-19e9a100-42a5-11ea-8669-255dcef71b78.png)
![Screenshot from 2020-01-29 14-37-42](https://user-images.githubusercontent.com/26549754/73357466-1a823780-42a5-11ea-99b3-b6ece6ccbc5b.png)
![Screenshot from 2020-01-29 14-38-04](https://user-images.githubusercontent.com/26549754/73357467-1a823780-42a5-11ea-9b36-ad4ece0d8303.png)

**Comment by aeschli** [comment]
@limbosounds Thanks, that was helpful.
Its bug #89600.
Because the Monokai Dimmed theme has no rule for 'variable`  the bug causes all variables to be rendered white.

Fix is on the way.

**Comment by limbosounds** [comment]
Seems like fixed in be0aca7

**Comment by aeschli** [comment]
Cool, thanks for the feedback
